packages/cli/src/ui/hooks/useKeypress.test.ts

```diff
@@ -258,4 +268,325 @@ describe('useKeypress', () => {
+        ctrl: true,
+        meta: false,
+        shift: false,
+        paste: false,
+      });
+
+      onKeypress.mockClear();
+
+      // Kitty protocol Ctrl+C
+      act(() => {
+        stdin.pressKey({ sequence: '\x1b[99;5u' });
+      });
+
+      expect(onKeypress).toHaveBeenCalledWith({
+        name: 'c',
+        ctrl: true,
+        meta: false,
+        shift: false,
+        paste: false,
+        sequence: '\x1b[99;5u',
+        kittyProtocol: true,
+      });
+    });
+  });
+
+  describe('VS Code Terminal Support', () => {
+    it('should convert backslash+return to Shift+Enter', () => {
+      const onKeypress = vi.fn();
+      renderHook(() => useKeypress(onKeypress, { isActive: true }));
+
+      // First send backslash
+      act(() => {
+        stdin.pressKey({
+          name: undefined,
+          sequence: '\\',
+          ctrl: false,
+          meta: false,
+          shift: false,
+        });
+      });
+
+      // Backslash should be buffered, not passed through
+      expect(onKeypress).not.toHaveBeenCalled();
+
+      // Then send return
+      act(() => {
+        stdin.pressKey({
+          name: 'return',
+          sequence: '\r',
+          ctrl: false,
+          meta: false,
+          shift: false,
+        });
+      });
+
+      // Should be converted to Shift+Enter
+      expect(onKeypress).toHaveBeenCalledWith({
+        name: 'return',
+        sequence: '\\\r',
+        ctrl: false,
+        meta: false,
+        shift: true,
+        paste: false,
+      });
+    });
+
+    it('should not convert non-backslash+return sequences', () => {
+      const onKeypress = vi.fn();
+      renderHook(() => useKeypress(onKeypress, { isActive: true }));
+
+      // Send 'a' then return
+      act(() => {
+        stdin.pressKey({
+          name: 'a',
+          sequence: 'a',
+          ctrl: false,
+          meta: false,
+          shift: false,
+        });
+      });
+
+      onKeypress.mockClear();
+
+      act(() => {
+        stdin.pressKey({
+          name: 'return',
+          sequence: '\r',
+          ctrl: false,
+          meta: false,
+          shift: false,
+        });
+      });
+
+      // Should not be converted
+      expect(onKeypress).toHaveBeenCalledWith({
+        name: 'return',
+        sequence: '\r',
+        ctrl: false,
+        meta: false,
+        shift: false,
+        paste: false,
+      });
+    });
+
+    it('should pass through backslash when not followed by return', () => {
+      const onKeypress = vi.fn();
+      renderHook(() =>
+        useKeypress(onKeypress, { isActive: true }),
+      );
+
+      // Send backslash
+      act(() => {
+        stdin.pressKey({
+          name: undefined,
+          sequence: '\\',
+          ctrl: false,
+          meta: false,
+          shift: false,
+        });
+      });
+
+      // Backslash should be buffered
+      expect(onKeypress).not.toHaveBeenCalled();
+
+      // Send 'a' after backslash
+      act(() => {
```

Comment:
can you also add a test where there is a backslash but not a follow up character. Seems like this ends up still working after an imperceptible timeout but would be good to verify that case is handled robustly as would be confusing to users otherwise

Reply:
Done